src/handlers/transfers/validator.js

```diff
@@ -99,14 +100,23 @@ const validateAmount = (amount) => {
   return true
 }
 
+// TODO: The following function should be moved into a re-usable common-shared-service at a later point
 // NOTE: This logic is based on v1.0 of the Mojaloop Specification as described in section 6.5.1.2
-const validateFulfilCondition = (fulfilment, condition) => {
-  // TODO: The following hashing code should be moved into a re-usable common-shared-service at a later point
+const fulfilmentToCondition = (fulfilment) => {
   var hashSha256 = Crypto.createHash('sha256')
-  var calculatedCondition = fulfilment
+  // var calculatedCondition = fulfilment // based on 6.5.1.2, the hash should be done on the decoded value as per the next line
```

Comment:
@elnyry @ksatya77 can you guys validate this please. It will impact how the condition is calculated from the fulfil.

There are two options here:
1. First Base64 decode the incoming Fulfil before calculating the Condition
2. Use the Base64 encoded incoming Fulfil to calculating the Condition

From my understanding of  `6.5.1.2` from the ML API Specification, the 1st option would be correct.

Let me know please.

Reply:
Lets go with option-1 .. I'll discuss with Murthy and get back if there is any issue